api/resources_portal/views/document_views.py

```diff
@@ -0,0 +1,378 @@
+filters can be found in the [Django-ES-DSL-DRF docs](https://django-elasticsearch-dsl-drf.readthedocs.io/en/0.17.1/filtering_usage_examples.html#filtering)
+
+There's an additional field in the response named `facets` that contain stats on the number of results per filter type.
+
+Example Requests:
+```
+?search=zebrafish
+?id=1
+?search=zebrafish&category=plasmid&has_publication=true
+?ordering=source_first_published
+```
+""",
+    ),
+)
+class MaterialDocumentView(DocumentViewSet):
+    """Defines search and filter fields for Materials"""
+
+    document = MaterialDocument
+    serializer_class = MaterialDocumentSerializer
+    pagination_class = ESLimitOffsetPagination
+
+    # Filter backends provide different functionality we want
+    filter_backends = [
+        FilteringFilterBackend,
+        OrderingFilterBackend,
+        DefaultOrderingFilterBackend,
+        CompoundSearchFilterBackend,
+        FacetedSearchFilterBackend,
+    ]
+    lookup_field = "id"
+
+    # Define search fields
+    search_fields = {
+        "title": {"boost": 10},
+        "category": None,
+        "contact_user": None,
+        "additional_metadata": None,
+        "pubmed_id": None,
+        "organism": None,
+        "additional_info": None,
+        "contact_name": None,
+        "contact_email": None,
+        "contact_user.published_name": None,
+        "publication_title": None,
+        "pre_print_doi": None,
+        "pre_print_title": None,
+        "citation": None,
+    }
+
+    # Define filtering fields
+    filter_fields = {
+        "id": {"field": "_id", "lookups": [LOOKUP_FILTER_RANGE, LOOKUP_QUERY_IN],},
+        "category": "category",
+        "organization": {"field": "name"},
+        "title": "title",
+        "organism": "organism",
+        "contact_user.email": "contact_user.email",
+        "contact_user.published_name": "contact_user.published_name",
+        "has_publication": "has_publication",
+        "has_pre_print": "has_pre_print",
+    }
+
+    # Define ordering fields
+    ordering_fields = {
+        "id": "id",
+        "title": "title.raw",
+        "category": "category.raw",
+        "created_at": "created_at",
+        "updated_at": "updated_at",
+    }
+
+    # Specify default ordering
+    ordering = (
+        "_score",
+        "updated_at",
+        "id",
+    )
+
+    faceted_search_fields = {
+        "category": {"field": "category", "facet": TermsFacet, "enabled": True},
+        "organism": {"field": "organism", "facet": TermsFacet, "enabled": True},
+        "contact_user.email": {"field": "contact_user.email", "facet": TermsFacet, "enabled": True},
+        "contact_user.published_name": {
+            "field": "contact_user.published_name",
+            "facet": TermsFacet,
+            "enabled": True,
+        },
+        "has_publication": {"field": "has_publication", "facet": TermsFacet, "enabled": True},
+        "has_pre_print": {"field": "has_pre_print", "facet": TermsFacet, "enabled": True},
```

Comment:
Would it be possible to group `has_publication` and `has_pre_print` into `publicaton_info` ?

Reply:
Resolved after conversation with david.